Tools/environment_install/install-ROS-ubuntu.sh

```diff
@@ -0,0 +1,253 @@
+# If you experience issues connecting to the keyserver, you can try substituting hkp://pgp.mit.edu:80 or hkp://keyserver.ubuntu.com:80 in the previous command.
+# Installation
+
+heading "${green}Updating apt${reset}"
+sudo apt update
+
+heading "${green}Installing ROS${reset}"
+# Here we loop through any packages passed on the command line
+# Install packages ...
+for package in "${packages[@]}"; do
+  sudo apt install $package -y
+done
+
+# This is where you might start to modify the packages being installed, i.e.# sudo apt install ros-${ROS_VERSION}-{package_name}
+# sudo apt install -y setpriv
+# sudo apt install -y ros-${ROS_VERSION}-robot-upstart
+# sudo apt install -y ros-${ROS_VERSION}-navigation
+
+# Install MAVROS and the geographic libs
+sudo apt install -y ros-${ROS_VERSION}-mavros
+
+wget  https://raw.githubusercontent.com/mavlink/mavros/master/mavros/scripts/install_geographiclib_datasets.sh
+sudo bash ./install_geographiclib_datasets.sh
+
+# install other needed packages
+sudo apt install build-essential cmake -y
+
+#
+# To find available packages:
+# apt-cache search ros-melodic
+# 
+# Initialize rosdep
+
+heading "${green}Installing rosdep${reset}"
+
+sudo apt install ${PYTHON_V}-rosdep -y
+# Certificates are messed up on earlier version Jetson for some reason
+# Do not know if it is an issue with the Xavier, test by commenting out
+# sudo c_rehash /etc/ssl/certs
+# Initialize rosdep
+
+heading "${green}Initializaing rosdep${reset}"
+
+sudo rosdep init || true
+# To find available packages, use:
+rosdep update
+# Use this to install dependencies of packages in a workspace
+# rosdep install --from-paths src --ignore-src --rosdistro=${ROS_DISTRO} -y
+# Environment Setup - Don't add /opt/ros/${ROS_VERSION}/setup.bash if it's already in bashrc
+if maybe_prompt_user "Do you want to add ROS_HOSTNAME and ROS_MASTER_URI to your .bashrc [N/y]?" ; then
+    heading "${green}Adding setup.bash, ROS_MASTER_URI and ROS_HOSTNAME to .bashrc ${reset}"
+    grep -q -F "ROS_HOSTNAME=$HOSTNAME.local" ~/.bashrc || echo "ROS_HOSTNAME=$HOSTNAME.local" >> ~/.bashrc
+    grep -q -F "ROS_MASTER_URI=http://$HOSTNAME.local:11311" ~/.bashrc || echo "ROS_MASTER_URI=http://$HOSTNAME.local:11311" >> ~/.bashrc
+    grep -q -F "source /opt/ros/${ROS_VERSION}/setup.bash" ~/.bashrc || echo "source /opt/ros/${ROS_VERSION}/setup.bash" >> ~/.bashrc
+    source ~/.bashrc
+fi
+
+heading "${green}Installing rosinstall tools${reset}"
+
+sudo apt install ${PYTHON_V}-rosinstall ${PYTHON_V}-rosinstall-generator ${PYTHON_V}-wstool ${PYTHON_V}-catkin-tools -y
+
+heading "${green}Installing Ardupilot-ROS workspace${reset}"
+
+if maybe_prompt_user "Add ardupilot-ws to your home folder [N/y]?" ; then
+    if [ ! -d $ROS_WS_ROOT ]; then
+        mkdir -p $ROS_WS_ROOT/src
+        pushd $ROS_WS_ROOT
+        catkin init
+        pushd src
+        git clone https://github.com/snktshrma/ardupilot_ros.git
```

Comment:
`
wget https://github.com/ArduPilot/companion/raw/master/Common/ROS/ap_navigation.zip
`
`unzip ap_navigation.zip`

Reply:
Why a zip a not gitted version ?

Reply:
Yes! Maybe we can have a separate repo for ap_navigation